### PR TITLE
Fix the problem of linux login during guest poweron

### PR DESCRIPTION
**Pull request by nanzhg**
Fix the needle match failure for open-vm-tools tests - https://openqa.suse.de/tests/14384793#step/esxi_open_vm_tools/61

- Related ticket: https://progress.opensuse.org/issues/159081
- Verification run: https://openqa.suse.de/tests/14423446

### CODE REVIEW

**Comment by alice-suse** [comment]
I find a weird thing here. From the verification job also the failure test jobs, https://openqa.suse.de/tests/14422019#step/disable_grub_timeout/8 has disabled the grub timeout. How can it be possible that grub2 isn't caught and it directly boots into system without clicking Enter on grub2 menu.

**Comment by nanzhg** [comment]
> I find a weird thing here. From the verification job also the failure test jobs, https://openqa.suse.de/tests/14422019#step/disable_grub_timeout/8 has disabled the grub timeout. How can it be possible that grub2 isn't caught and it directly boots into system without clicking Enter on grub2 menu.

It was enabled in add_serial_console test module again. :-)
https://openqa.suse.de/tests/14422019#step/add_serial_console/24

**Comment by alice-suse** [comment]
> > I find a weird thing here. From the verification job also the failure test jobs, https://openqa.suse.de/tests/14422019#step/disable_grub_timeout/8 has disabled the grub timeout. How can it be possible that grub2 isn't caught and it directly boots into system without clicking Enter on grub2 menu.
> 
> It was enabled in add_serial_console test module again. :-) https://openqa.suse.de/tests/14422019#step/add_serial_console/24

Aha, good catch! Then why not enlarge the timeout? I think claiming grub2 is useful for debug sometimes.

**Comment by nanzhg** [comment]
> > > I find a weird thing here. From the verification job also the failure test jobs, https://openqa.suse.de/tests/14422019#step/disable_grub_timeout/8 has disabled the grub timeout. How can it be possible that grub2 isn't caught and it directly boots into system without clicking Enter on grub2 menu.
> > 
> > 
> > It was enabled in add_serial_console test module again. :-) https://openqa.suse.de/tests/14422019#step/add_serial_console/24
> 
> Aha, good catch! Then why not enlarge the timeout? I think claiming grub2 is useful for debug sometimes.

Good idea! Updated, thanks.

File: tests/installation/add_serial_console.pm [hunk 1]
```diff
@@ -33,8 +33,8 @@ sub run {
     assert_script_run('sed -ie \'s/GRUB_TERMINAL.*//\' /etc/default/grub');
     # VMware needs to always stop in Grub and wait for interaction.
     # Migration seems to reset it.
-    if (get_var('KEEP_GRUB_TIMEOUT')) {
-        assert_script_run('sed -ie \'s/GRUB_TIMEOUT.*/GRUB_TIMEOUT=8/\' /etc/default/grub');
+    if (get_var('GRUB_TIMEOUT')) {
+        assert_script_run('sed -ie \'s/GRUB_TIMEOUT.*/GRUB_TIMEOUT=' . get_var('GRUB_TIMEOUT') . '/\' /etc/default/grub');
```
Comment (alice-suse) [inline tests/installation/add_serial_console.pm:37]: Allowing customized GRUB_TIMEOUT is great. But will deleting KEEP_GRUB_TIMEOUT be problematic for other squads' tests which are setting KEEP_GRUB_TIMEOUT only? Maybe some minor change with backward compatibility will survive all.

Comment (alice-suse) [reply tests/installation/add_serial_console.pm:37]: Okay, if you are sure that our test is the only user, then it will be fine.

**Review by alice-suse — APPROVED** [review]
LGTM

**Pull request merged**